tests/cpp/test_matmul_aten_evaluation.cpp

```diff
@@ -31,6 +31,21 @@ class MatmulATenEvaluationTest : public NVFuserTest {
       optimization_guard_;
 };
 
+using MatmulNodeParamType =
+    std::tuple<std::vector<int64_t>, std::vector<int64_t>>;
+
+class ATenNodesParametrizedTest
+    : public NVFuserFixtureParamTest<MatmulNodeParamType> {
+ protected:
+  // Allocation order set by the pass breaks matmul tests
+  // see issue https://github.com/NVIDIA/Fuser/issues/1810
+  ATenNodesParametrizedTest() : optimization_guard_(false) {}
+
+ private:
```

Comment:
Since this depends on aten evaluation, should these tests also add a `DisableOptionsGuard` that ensures `matmul_expr_eval` is unset?

Reply:
I was going to remove setting that flag within the matmul scheduler once I added the ATen scheduler in the next PR.

I don't see a reason to keep this option anymore.

For these tests in particular, they would be unaffected since the outputs are marked for expression evaluator in the ATen scheduler anyway, skipping compilation and would behave the same way as the existing tests (using pattern matching)

Reply:
Right, we can reuse this option to disable the scheduler. I can add that in the scheduler PR: #2209 (Draft)